lib/features/email/presentation/email_controller.dart

```diff
@@ -263,6 +333,17 @@ class EmailController extends BaseController with AppLoaderMixin {
   }
 
   void _getEmailContentSuccess(GetEmailContentSuccess success) {
+    if(presentationEmailsLoaded.length > ThreadConstants.defaultLimit.value.toInt()) {
+      presentationEmailsLoaded.removeFirst();
+    }
+    final emailLoaded = EmailLoaded(
+      success.emailContents.toList(),
+      success.emailContentsDisplayed.toList(),
+      success.attachments.toList(),
+      success.emailCurrent,
+    );
+    presentationEmailsLoaded.removeWhere((e) => e == emailLoaded);
```

Comment:
Why you need to remove the same email?
IMO, should be `removeWhere((e) => e.id == emailLoadedId)`.